internal/controller/progressive/progressive.go

```diff
@@ -410,10 +423,20 @@ func AssessUpgradingPipelineType(
 		return apiv1.AssessmentResultUnknown, replicasFailureReason, err
 	}
 
+	if upgradingObjectStatus.Phase == "Failed" || !healthyConditions || !healthyReplicas {
+		failureReason := CalculateFailureReason(replicasFailureReason, upgradingObjectStatus.Phase, failedCondition)
+		return apiv1.AssessmentResultFailure, failureReason, nil
+	}
```

Comment:
sorry if I may have said the wrong thing, but let's actually make sure the logging that's just below this occurs prior to this